nginx-directory/configfile.libsonnet

```diff
@@ -11,8 +11,10 @@ function(services) {
         return 302 %(url)s;
       ||| % service
       else |||
-        proxy_pass      %(url)s$2$is_args$args;
```

Comment:
Have we lost the handling of query parameters here? If I understand HTTP redirects correctly the client will request the exact Location URL and that needs to include the query parameters if we wish to pass them to redirect.

Reply:
The query parameters do remain on the URL visible in the browser, and the logs from my webserver on the other end of the proxy look like this:
```34.122.9.230 - - [26/Jul/2021:10:07:11 -0400] "GET /?parameter=hello HTTP/1.0" 200 1758 "-" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.164 Safari/537.36"
```

Reply:
It seems that with `location` being in non-regex mode, it is able to automatically keep everything.
Unfortunately when in regex mode, the `$2` in this line caused the `Space%20Separated` to turn into `Space Separated`.

Reply:
Thanks for confirming. I don't know NGINX very well but that observational evidence along with the explanation of the regexp behavior makes sense and gives me confidence that we aren't introducing any regression.

Reply:
One thing I've noticed while experimenting with this change is that this comment will be repeated for every entry in the NGINX directory. I wonder if we should consider moving the comment outside of the actual configuration string to avoid this. The downside would be that the comment would need more context and would be more likely to be neglected in the future.